session_0/word_processing.rb

```diff
@@ -34,11 +55,35 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  i = 0
+  while ((word_1[i] == word_2[i]) && (i < word_1.length) && (i < word_2.length)) do
```

Comment:
This seems a bit too tedious,try using slice constructs

Reply:
I've cleaned up the code a bit but I do not understand how slice can be used to simplify it any further